dcos/http.py

```diff
@@ -167,9 +173,12 @@ def request(method,
         return response
     elif response.status_code == 401:
         if auth_token is not None:
-            msg = ("Your core.dcos_acs_token is invalid. "
-                   "Please run: `dcos auth login`")
-            raise DCOSAuthenticationException(msg)
+            if prompt_login:
```

Comment:
we should call the auth method whether or not there is an auth token. The original condition is for the message we show to the user.